xilem_core/src/views/orphan.rs

```diff
@@ -0,0 +1,189 @@
+            }
+
+            fn message(
+                &self,
+                view_state: &mut Self::ViewState,
+                id_path: &[ViewId],
+                message: DynMessage,
+                app_state: &mut State,
+            ) -> MessageResult<Action> {
+                Context::message(self, view_state, id_path, message, app_state)
+            }
+        }
+    };
+}
+
+macro_rules! impl_as_orphan_view_for {
+    ($ty: ty) => {
+        impl<State, Action, Context> View<State, Action, Context> for $ty
+        where
+            Context: AsOrphanView<$ty, State, Action>,
+        {
+            type Element = <<Context as AsOrphanView<$ty, State, Action>>::V as View<State, Action, Context>>::Element;
+            type ViewState = <<Context as AsOrphanView<$ty, State, Action>>::V as View<State, Action, Context>>::ViewState;
+
+            fn build(&self, ctx: &mut Context) -> (Self::Element, Self::ViewState) {
+                <<Context as AsOrphanView<$ty, State, Action>>::V as View<State, Action, Context>>
+                    ::build(&Context::as_view(self), ctx)
+            }
+
+            fn rebuild<'el>(
+                &self,
+                prev: &Self,
+                view_state: &mut Self::ViewState,
+                ctx: &mut Context,
+                element: Mut<'el, Self::Element>,
+            ) -> Mut<'el, Self::Element> {
+                <<Context as AsOrphanView<$ty, State, Action>>::V as View<State, Action, Context>>::rebuild(
+                    &Context::as_view(self),
+                    &Context::as_view(prev),
+                    view_state,
+                    ctx,
+                    element,
+                )
+            }
+
+            fn teardown(
+                &self,
+                view_state: &mut Self::ViewState,
+                ctx: &mut Context,
+                element: <Self::Element as ViewElement>::Mut<'_>,
+            ) {
+                <<Context as AsOrphanView<$ty, State, Action>>::V as View<State, Action, Context>>
+                    ::teardown(&Context::as_view(self), view_state, ctx, element);
+            }
+
+            fn message(
+                &self,
+                view_state: &mut Self::ViewState,
+                id_path: &[ViewId],
+                message: DynMessage,
+                app_state: &mut State,
+            ) -> MessageResult<Action> {
+                <<Context as AsOrphanView<$ty, State, Action>>::V as View<State, Action, Context>>::message(
+                    &Context::as_view(self),
+                    view_state,
+                    id_path,
+                    message,
+                    app_state,
+                )
+            }
+        }
+    };
+}
+
+// string impls
+impl_as_orphan_view_for!(&'static str);
+#[cfg(feature = "std")]
+impl_orphan_view_for!(String);
+#[cfg(feature = "std")]
+impl_as_orphan_view_for!(std::borrow::Cow<'static, str>);
+// Why does the following not work, but the `Cow` impl does??
```

Comment:
I presume this would be because we already have an impl for `Arc<T>`, and the compiler (for some reason) can't reason that `str` is not `View`.

Reply:
Ahh, that makes sense, thanks for the hint.